src/file.js

```diff
@@ -157,18 +203,27 @@ Sk.builtin.file.prototype["truncate"] = new Sk.builtin.func(function (self, size
 });
 
 Sk.builtin.file.prototype["write"] = new Sk.builtin.func(function (self, str) {
-    if (Sk.filewrite) {
-        if (self.closed) {
-            throw new Sk.builtin.ValueError("I/O operation on closed file");
-        }
+    var mode = Sk.ffi.remapToJs(self.mode);
+    if (self.mode.v === "w" || self.mode.v === "wb" || self.mode.v === "a" || self.mode.v === "ab") {
+        if (Sk.filewrite) {
+            if (self.closed) {
+                throw new Sk.builtin.ValueError("I/O operation on closed file");
+            }
 
-        if (self.mode.v === "w" || self.mode.v === "wb" || self.mode.v === "a" || self.mode.v === "ab") {
-            Sk.filewrite(self, str);
+            if (self.fileno === 1) {
```

Comment:
@brianpmarks as you can see here it'll only call `Sk.filewrite` when it's not writing to the `/dev/stdout`